### DISCUSSION TITLE
Complex version parsing?

### DISCUSSION THREAD

**Discussion by jbergler** [renovatebot/renovate#10039] [Request Help]
Hey there, hoping that someone might help me feel a little less dumb as I beat my head against a wall trying to deal with a project that has dumb versioning formats.

All three of these versions appear for a particular docker container.

```
3.0.6.1196-ls105
5.14-version-2.0.0.5344
develop-3.0.6.1233-ls169
```

And I had hoped to tackle this with something like the following, so that I end up with develop-3.0.6.1233-ls169 as the selected version

```
      "versionScheme": "regex:^(?<prerelease>develop-)?(?<compatibility>.*-)?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+\\.\\d+(-\\w+)?)$",                                                           
      "ignoreUnstable": false
```

But I'm now running into the challenge that when renovate parses the version numbers, patch is `NaN` and so the whole thing explodes.

The main alternative I'm considering is using `versioning: loose` and `allowedVersions: /^develop-/` or something, but I was hoping to be able to find a middle ground that still gave me the flexibility to detect major/minor/patch level changes and apply further config based on that.

Any ideas?

**Comment by rarkins** [comment]
It seems like a pretty hard question.

First of all, you want to support only `develop-` versions?

**Reply by jbergler** [reply]
I think I end up toggling between the `/^develop-\d+` and the normal `/^\d+` versions a bit over time (where I totally accept that I need to make a config change to switch between these)

**Reply by rarkins** [reply]
OK. The next thing is that there are *four* parts to the `develop-` version example you give, plus a `-ls169` suffix. We don't natively support four-part versions yet. Do you expect the third part or the fourth part to change, or both? And what is the significance of the suffix?

**Reply by jbergler** [reply]
`3.0.6.1233` represents the version of the actual software itself, while the `ls169` suffix represents the base container (which gets updated when the OS gets patched but the version of the software doesn't change).

**Reply by rarkins** [reply]
I think you'd need #4860 to be implemented in order to properly support 4-part versions

**Reply by jbergler** [reply]
Understood, thanks for the help.

**Reply by viceice** [reply]
For those samples 4 parts are not enough. I see. 5 version parts. 😕
